mailbox/tools/indexer/src/main/java/org/apache/mailbox/tools/indexer/ReIndexerImpl.java

```diff
@@ -39,15 +45,28 @@
 public class ReIndexerImpl implements ReIndexer {
 
     private final ReIndexerPerformer reIndexerPerformer;
+    private final MailboxManager mailboxManager;
+    private final MailboxSessionMapperFactory mapperFactory;
 
     @Inject
-    public ReIndexerImpl(ReIndexerPerformer reIndexerPerformer) {
+    public ReIndexerImpl(ReIndexerPerformer reIndexerPerformer, MailboxManager mailboxManager, MailboxSessionMapperFactory mapperFactory) {
         this.reIndexerPerformer = reIndexerPerformer;
+        this.mailboxManager = mailboxManager;
+        this.mapperFactory = mapperFactory;
     }
 
     @Override
-    public Task reIndex(MailboxPath path) {
-        return new SingleMailboxReindexingTask(reIndexerPerformer, path);
+    public Task reIndex(MailboxPath path) throws MailboxException {
+        MailboxSession mailboxSession = mailboxManager.createSystemSession("ReIndexingImap");
+        Mailbox mailbox = mapperFactory.getMailboxMapper(mailboxSession).findMailboxByPath(path);
+        return new SingleMailboxReindexingTask(reIndexerPerformer, mailbox);
+    }
+
+    @Override
+    public Task reIndex(MailboxId mailboxId) throws MailboxException {
+        MailboxSession mailboxSession = mailboxManager.createSystemSession("ReIndexingImap");
+        Mailbox mailbox = mapperFactory.getMailboxMapper(mailboxSession).findMailboxById(mailboxId);
```

Comment:
mailboxManager.search ?

Reply:
Not possible: right would be denied as you don't know yet what user that mailbox does belong to.

We miss an always allowed session...